### PR TITLE
[KED-1497] Add in bandit for security scanning as a pre-commit hook

### PR DESCRIPTION
**Pull request by mzjp2**
## Description

Add bandit security scanning to the repo.

## Development notes

Added a make command and added a step as part of the circleci config

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -70,6 +70,11 @@ utils:
     command: |
       make secret-scan
 
+  run_security_scan: &run_security_scan
+    name: Run security scan
+    command: |
+      make security-scan
```
Comment (DmitriiDeriabinQB) [inline .circleci/config.yml]:
```suggestion
    command: make security-scan
```

Comment (mzjp2) [reply .circleci/config.yml]: Should I change all the other instances of it (for secret-scan and pylint) too?

Comment (DmitriiDeriabinQB) [reply .circleci/config.yml]: Yeah if you want to 🙂
it's super minor, just saw it in diff, hence the comment)

Comment (mzjp2) [reply .circleci/config.yml]: Done :)

**Pull request merged**